src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/ClassWriter.java

```diff
@@ -1758,10 +1763,13 @@ public void writeClassFile(OutputStream out, ClassSymbol c)
         acount += writeExtraAttributes(c);
 
         poolbuf.appendInt(JAVA_MAGIC);
+        boolean markedPreview;
         if (preview.isEnabled() && preview.usesPreview(c.sourcefile)) {
             poolbuf.appendChar(ClassFile.PREVIEW_MINOR_VERSION);
+            markedPreview = true;
         } else {
             poolbuf.appendChar(target.minorVersion);
+            markedPreview = target.minorVersion == ClassFile.PREVIEW_MINOR_VERSION;
```

Comment:
I don't think `target.minorVersion` can ever be preview?

Reply:
IIRC, `forcePreview` makes use of _any_ feature behave as if they were preview features, meaning you will get classfile version pollution. But it doesn't alter Target. So yes, the code you suggest should be correct.